src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final ColorSupplier color = new ColorSupplier();
+    private static final Random random = new Random();
+    private static final FigureList[] figures = FigureList.values();
+    private static final double RANDOM_MAX_VALUE = 5.1;
+
+    public Figure getRandomFigure() {
+        Figure figure = null;
+
+        switch (randomiseFigureType()) {
+            case CIRCLE:
+                figure = new Circle(color.getRandomColor(), random.nextDouble(RANDOM_MAX_VALUE));
+                break;
+            case ISOSCELES_TRAPEZOID:
+                figure = new IsoscelesTrapezoid(color.getRandomColor(),
+                        random.nextDouble(RANDOM_MAX_VALUE), random.nextDouble(RANDOM_MAX_VALUE),
+                        random.nextDouble(RANDOM_MAX_VALUE));
+                break;
+            case RIGHT_TRIANGLE:
+                figure = new RightTriangle(color.getRandomColor(),
+                        random.nextDouble(RANDOM_MAX_VALUE), random.nextDouble(RANDOM_MAX_VALUE));
+                break;
+            case RECTANGLE:
+                figure = new Rectangle(color.getRandomColor(), random.nextDouble(RANDOM_MAX_VALUE),
+                        random.nextDouble(RANDOM_MAX_VALUE));
+                break;
+            case SQUARE:
+                figure = new Square(color.getRandomColor(), random.nextDouble(RANDOM_MAX_VALUE));
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. Also, consider extracting the logic to create specific figures into separate private methods to reduce the complexity of the 'getRandomFigure()' method.